### PR TITLE
fix: add payload to navigation event to resolve navigation loop

### PR DESCRIPTION
**Pull request by anninowak**

### CODE REVIEW

File: src/app/app.module.ts [hunk 1]
```diff
@@ -181,11 +181,15 @@ window.history.replaceState = (data: any, unused: string, url?: string) => {
...
-    }
+    } satisfies PayloadNavigatedEvent
   })
   if (!isInitialPageLoad) {
-    history.push(url ?? '')
-    history = history.slice(-100)
+    if (history.length === 0) {
+      history.push(url ?? '')
+    } else {
+      let length = history.length - 1
```
Comment (SchettlerKoehler) [inline src/app/app.module.ts:190]: This is not length. It is lastIndex

File: src/app/app.module.ts [hunk 2]
```diff
@@ -181,11 +181,15 @@ window.history.replaceState = (data: any, unused: string, url?: string) => {
...
       isFirst: history.length === 0,
       history
-    }
+    } satisfies PayloadNavigatedEvent
   })
   if (!isInitialPageLoad) {
-    history.push(url ?? '')
-    history = history.slice(-100)
+    if (history.length === 0) {
+      history.push(url ?? '')
```
Comment (SchettlerKoehler) [inline src/app/app.module.ts:188]: If the length is 0 I would do nothing

File: src/app/app.module.ts [hunk 3]
```diff
@@ -4,7 +4,7 @@ import { BrowserModule } from '@angular/platform-browser'
 import { BrowserAnimationsModule } from '@angular/platform-browser/animations'
 import { Router, RouterModule } from '@angular/router'
 import { MissingTranslationHandler, TranslateLoader, TranslateModule } from '@ngx-translate/core'
-
+import { PayloadNavigatedEvent } from '@onecx/integration-interface'
```
Comment (SchettlerKoehler) [inline src/app/app.module.ts:7]: NavigatedEventPayload

**Review by SchettlerKoehler — DISMISSED** [review]

File: src/app/app.module.ts (hunk 3 above)
Comment (anninowak) [reply src/app/app.module.ts:7]: libs were already released with this name.
Change it in the future?

File: src/app/app.module.ts (hunk 2 above)
Comment (anninowak) [reply src/app/app.module.ts:188]: clarified with you

File: src/app/app.module.ts (hunk 1 above)
Comment (anninowak) [reply src/app/app.module.ts]: changed

File: src/app/app.module.ts (hunk 3 above)
Comment (SchettlerKoehler) [reply src/app/app.module.ts:7]: Please rename it and create a deprecated type acting as an alias 
```
@deprecated
type PayloadNavigatedEvent  = NavigatedEventPayload
```
With this, both are working but you are saying that `NavigatedEventPayload` is the variant you want to have in the future

**Review by milanhorvath — APPROVED** [review]

**Pull request merged**